src/Consumer/Consumer.php

```diff
@@ -52,62 +53,74 @@ public function getCallback(): callable
 
 	public function consume(?int $maxSeconds = null, ?int $maxMessages = null): void
 	{
+		$this->maxMessages = $maxMessages;
 		$channel = $this->queue->getConnection()->getChannel();
 
 		if ($this->prefetchSize !== null || $this->prefetchCount !== null) {
 			$channel->qos($this->prefetchSize ?? 0, $this->prefetchCount ?? 0);
 		}
 
 		$channel->consume(
-			function (Message $message, Channel $channel, Client $client) use ($maxMessages): void {
+			function (Message $message, Channel $channel, Client $client): void {
 				$result = call_user_func($this->callback, $message);
 
-				switch ($result) {
-					case IConsumer::MESSAGE_ACK:
-						// Acknowledge message
-						$channel->ack($message);
+				$this->sendResponse($message, $channel, $result, $client);
 
-						break;
+				if ($this->isMaxMessages()) {
+					$client->stop();
+				}
+			},
+			$this->queue->getName()
+		);
 
-					case IConsumer::MESSAGE_NACK:
-						// Message will be requeued
-						$channel->nack($message);
+		$channel->getClient()->run($maxSeconds);
+	}
 
-						break;
+	protected function sendResponse(Message $message, Channel $channel, int $result, Client $client): void
+	{
+		switch ($result) {
+			case IConsumer::MESSAGE_ACK:
+				// Acknowledge message
+				$channel->ack($message);
 
-					case IConsumer::MESSAGE_REJECT:
-						// Message will be discarded
-						$channel->reject($message, false);
+				break;
 
-						break;
+			case IConsumer::MESSAGE_NACK:
+				// Message will be requeued
+				$channel->nack($message);
 
-					case IConsumer::MESSAGE_REJECT_AND_TERMINATE:
-						// Message will be discarded
-						$channel->reject($message, false);
-						$client->stop();
+				break;
 
-						break;
+			case IConsumer::MESSAGE_REJECT:
+				// Message will be discarded
+				$channel->reject($message, false);
 
-					case IConsumer::MESSAGE_ACK_AND_TERMINATE:
-						// Acknowledge message and terminate
-						$channel->ack($message);
-						$client->stop();
+				break;
 
-						break;
+			case IConsumer::MESSAGE_REJECT_AND_TERMINATE:
+				// Message will be discarded
+				$channel->reject($message, false);
+				$client->stop();
 
-					default:
-						throw new \InvalidArgumentException(
-							"Unknown return value of consumer [{$this->name}] user callback"
-						);
-				}
+				break;
 
-				if ($maxMessages !== null && ++$this->messages >= $maxMessages) {
```

Comment:
This line was lost from previous version. Could you please alter the code to preserve current functionality?

If we are not 100% sure what all the previous code does, we don't need to change any Consumer code and only put the new one in BulkConsumer. I will leave that up 2 you

Reply:
We have fixed that, it was lost in refactoring code to de-duplicate it.